### PR TITLE
add easyblock for building and installing TINKER

### PR DESCRIPTION
**Pull request by boegel**

### CODE REVIEW

**Comment by JensTimmerman** [comment]
looks ok to me

**Comment by boegel** [comment]
broken test reported via https://github.com/jayponder/tinker/issues/7

File: easybuild/easyblocks/t/tinker.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        if comp_dir is None:
+            self.log.error("Failed to determine compiler directory for %s (known: %s)" % (comp_fam, comp_dirs))
+
+        self.build_subdir = os.path.join(os_dir, comp_dir)
+        self.log.info("Using build scripts from %s subdirectory" % self.build_subdir)
+
+        # patch 'link.make' script to use FFTW provided via EasyBuild
+        link_make_fp = os.path.join(self.cfg['start_dir'], self.build_subdir, 'link.make')
+        for line in fileinput.input(link_make_fp, inplace=1, backup='.orig'):
+            line = re.sub(r"libfftw3_threads.a libfftw3.a", r"-L$EBROOTFFTW/lib -lfftw3_threads -lfftw3", line)
```
Comment (wpoely86) [inline easybuild/easyblocks/t/tinker.py]: it's unclear to me, but if this a makefile you're patching, you need `$(EBROOTFFTW)`

File: easybuild/easyblocks/t/tinker.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+                os.chdir(testdir)
+            except OSError, err:
+                self.log.error("Failed to move to %s to run tests: %s" % (testdir, err))
+
+            # run all tests via the provided 'run' scripts
+            tests = glob.glob(os.path.join(testdir, '*.run'))
+            # gpcr takes too logn (~1h), ifabp fails due to input issues (?)
+            tests = [t for t in tests if not (t.endswith('gpcr.run') or t.endswith('ifabp.run'))]
+            for test in tests:
+                run_cmd(test)
```
Comment (wpoely86) [inline easybuild/easyblocks/t/tinker.py]: `simple=True`?

File: easybuild/easyblocks/t/tinker.py [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+            'files': ['tinker/source/libtinker.a'],
+            'dirs': ['tinker/bin'],
+        }
+        super(EB_TINKER, self).sanity_check_step(custom_paths=custom_paths)
+
+    def make_module_req_guess(self):
+        """Custom guesses for module file prepend-path statements."""
+        guesses = super(EB_TINKER, self).make_module_req_guess()
+        guesses['PATH'].append(os.path.join('tinker', 'bin'))
+        guesses['LIBRARY_PATH'].append(os.path.join('tinker', 'source'))
```
Comment (wpoely86) [inline easybuild/easyblocks/t/tinker.py]: no `LD_LIBRARY_PATH`?

File: easybuild/easyblocks/t/tinker.py (hunk 1 above)
Comment (boegel) [inline easybuild/easyblocks/t/tinker.py]: it's not a `Makefile`, it's a script, basically

see https://github.com/jayponder/tinker/blob/master/linux/intel/link.make

File: easybuild/easyblocks/t/tinker.py (hunk 2 above)
Comment (boegel) [inline easybuild/easyblocks/t/tinker.py]: doesn't really matter, does it?

it checks the exit code, and nicely logs the output, so the defaults are fine

File: easybuild/easyblocks/t/tinker.py (hunk 3 above)
Comment (boegel) [inline easybuild/easyblocks/t/tinker.py]: only a static library, so setting `$LD_LIBRARY_PATH` is useless

**Comment by boegel** [comment]
Thanks for the review @JensTimmerman & @wpoely86!

**Comment by boegel** [comment]
Going in, thanks again for reviewing!

**Comment by boegel** [comment]
I was putting this on hold until I had a working test report, which is now there, see https://github.com/hpcugent/easybuild-easyconfigs/pull/1465#issuecomment-78518164.

**Pull request merged**